force-app/main/default/classes/SOQL.cls

```diff
@@ -1365,27 +1369,40 @@ public inherited sharing class SOQL implements Queryable {
     }
 
     private class Mock {
-        private final Map<String, List<SObject>> mocks = new Map<String, List<SObject>>();
+        private final Map<String, SObject> sObjectMocks = new Map<String, SObject>();
+        private final Map<String, List<SObject>> sObjectsMocks = new Map<String, List<SObject>>();
         private final Map<String, Integer> countMocks = new Map<String, Integer>();
 
+        public void setMock(String mockId, SObject record) {
+            sObjectMocks.put(mockId, record);
+        }
+
         public void setMock(String mockId, List<SObject> records) {
```

Comment:
It duplicates code, you can use list for singular object types as well

Reply:
Yep!